plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseDbLink.java

```diff
@@ -0,0 +1,149 @@
+
+    private String userName; // null for public, user_id for private
+    private int linkId;
+    private int userMode; // 0:public, 1: private
+    private String remoteUserId;
+    private String targetName;
+    private int linkType; // Heterogeneous Link or Homogeneous Link: Only Heterogeneous since 6.5.1
+    private Timestamp created;
+    private Timestamp lastDdlTime;
+
+    protected AltibaseDbLink(GenericStructContainer parent, JDBCResultSet resultSet) {
+        super(parent, 
+                JDBCUtils.safeGetString(resultSet, "LINK_NAME"), 
+                JDBCUtils.safeGetLong(resultSet, "LINK_OID"),
+                true);
+
+        userName = JDBCUtils.safeGetString(resultSet, "USER_NAME");
+        linkId = JDBCUtils.safeGetInt(resultSet, "LINK_ID");
+        userMode = JDBCUtils.safeGetInt(resultSet, "USER_MODE");
+        remoteUserId = JDBCUtils.safeGetString(resultSet, "REMOTE_USER_ID");
+        linkType = JDBCUtils.safeGetInt(resultSet, "LINK_TYPE");
+        targetName = JDBCUtils.safeGetString(resultSet, "TARGET_NAME");
+        created = JDBCUtils.safeGetTimestamp(resultSet, "CREATED");
+        lastDdlTime = JDBCUtils.safeGetTimestamp(resultSet, "LAST_DDL_TIME");
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 2)
+    public int getLinkId() {
+        return linkId;
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 3)
+    public long getLinkOid() {
+        return getObjectId();
+    }
+
+    @Nullable
+    @Property(viewable = true, order = 4)
+    public String getUserName() {
+        return userName;
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 5)
+    public String getRemoteUserId() {
+        return remoteUserId;
+    }
+
+
+    @NotNull
+    @Property(viewable = true, order = 6)
+    public String getTargetName() {
+        return targetName;
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 7)
+    public String getUserMode() {
+        return isPublic() ? "Public" : "Private";
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 8)
+    public String getLinkType() {
+        return (linkType == 0) ? "Heterogeneous" : "Homogeneous";
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 10)
+    public Timestamp getCreated() {
+        return created;
+    }
+
+    @NotNull
+    @Property(viewable = true, order = 11)
+    public Timestamp getLastDdlTime() {
+        return lastDdlTime;
+    }
+
+    private boolean isPublic() {
+        return (userMode == 0);
+    }
+
+    @Override
+    public String getObjectDefinitionText(DBRProgressMonitor monitor, Map<String, Object> options) throws DBException {
+        if (CommonUtils.isEmpty(ddl)) {
+            options.put("SCHEMA", isPublic() ? "PUBLIC" : this.getParentObject().getName());
+            ddl = ((AltibaseMetaModel) getDataSource().getMetaModel()).getDbLinkDDL(monitor, this, options);
+        }
+
+        return (CommonUtils.isEmpty(ddl)) ? "" : ddl + ";";
```

Comment:
Why don't you want to append `;` in the `if` above?

Reply:
This takes into account cases where creating DDL fails due to exceptional circumstances while fetching from DB. Although it is very unlikely, I implemented it this way because I thought there would be no need to add ';' to the empty DDL string if this case occurred.

Reply:
I'm suggesting to directly append the semicolon to the buffer the first time your read it to avoid redundant copies introduced by concatenation.

Reply:
Okay, since it's actually very unlikely to happen, I'll move the semicolon inside the if statement as you recommended.

Thanks for your review :)